controlplane/internal/controller/agentcontrolplane_controller.go

```diff
@@ -123,6 +128,17 @@ func (r *AgentControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		controllerutil.AddFinalizer(acp, acpFinalizer)
 	}
 
+	acpVersion, err := semver.NewVersion(acp.Spec.Version)
+	if err != nil {
+		log.Error(err, "invalid version", "version", acp.Spec.Version)
+		return ctrl.Result{}, err
+	}
+	if acpVersion.LessThan(*minVersion) {
+		conditions.MarkFalse(acp, controlplanev1alpha1.MachinesCreatedCondition, controlplanev1alpha1.MachineGenerationFailedReason,
+			clusterv1.ConditionSeverityError, fmt.Sprintf("version %v is not supported", acp.Spec.Version))
```

Comment:
```suggestion
			clusterv1.ConditionSeverityError, fmt.Sprintf("openshift version %v is not supported, must be greater than 4.14.0", acp.Spec.Version))
```

Minor: maybe a good idea to let them know the version of what and what's expected?